sycl/source/device_selector.cpp

```diff
@@ -158,26 +120,35 @@ int default_selector::operator()(const device &dev) const {
   // Since we deprecate SYCL_BE and SYCL_DEVICE_TYPE,
   // we should not disallow accelerator to be chosen.
   // But this device type gets the lowest heuristic point.
-  if (dev.is_accelerator())
+  if (dev.is_accelerator()) {
     Score += 75;
 
+    // if SYCL_DEVICE_FILTER is used raise priority of accelerator device to
+    // select it when it is explicitely mentioned there. Otherwise host device
+    // which has higher priority and is always present is selected.
+    auto FilterList = detail::SYCLConfig<detail::SYCL_DEVICE_FILTER>::get();
+    if (FilterList) {
+      bool ForceAcc = false;
+      for (const detail::device_filter &Filter : FilterList->get()) {
+        if ((Filter.Backend == backend::opencl ||
+             Filter.Backend == backend::all) &&
+            Filter.DeviceType == info::device_type::accelerator) {
+          ForceAcc = true;
+        }
```

Comment:
I would like you to think about the case where the user sets SYCL_DEVICE_FILTER=5, where sycl-ls listed ACC device at the index 5.
We need to give a bonus point for the ACC device, but this test fails to.
How do you handle this?

Reply:
reworked device selection